components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyConnectionFactory.java

```diff
@@ -100,19 +108,22 @@ private synchronized void bootstrap(ConnectionSettings connectionSettings) {
         }
     }
 
+    public CompletableFuture<Void> close() {
```

Comment:
Perhaps we should have a utility method somewhere for turning netty futures into `CompletableFuture`s somewhere, to avoid mixing it with this Styx-specific code.

Reply:
I could actually add one in `styx-common`.